SmartDeviceLink/private/SDLSoftButtonReplaceOperation.m

```diff
@@ -103,7 +103,7 @@ - (void)start {
 - (void)sdl_uploadInitialStateImagesWithCompletionHandler:(void (^)(void))handler {
     NSMutableArray<SDLArtwork *> *initialStatesToBeUploaded = [NSMutableArray array];
     for (SDLSoftButtonObject *object in self.softButtonObjects) {
-        if ([self sdl_artworkNeedsUpload:object.currentState.artwork]) {
+        if (self.fileManager != nil && [self.fileManager fileNeedsUpload:object.currentState.artwork] && self.softButtonCapabilities.imageSupported.boolValue) {
```

Comment:
```suggestion
        if ([self.fileManager fileNeedsUpload:object.currentState.artwork]) {
```

Everything else here is unnecessary, you check if images are supported near the top of `start`